oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -184,6 +187,31 @@ public static GoogleCredentials fromStream(
             fileType, USER_FILE_TYPE, SERVICE_ACCOUNT_FILE_TYPE));
   }
 
+  /**
+   * Creates a credential with quota project from environment if present
+   *
+   * @param environmentProvider a provider used to get environment values
+   * @return credential with quota project from envoronment
+   */
+  public GoogleCredentials createWithQuotaProject(EnvironmentProvider environmentProvider) {
+    String quotaProjectFromEnv = environmentProvider.getEnv(QUOTA_PROJECT_ENV_VAR);
+
+    if (quotaProjectFromEnv != null && !quotaProjectFromEnv.isEmpty()) {
+      return this.toBuilder().setQuotaProjectId(quotaProjectFromEnv).build();
+    }
+
+    return this;
+  }
+
+  /**
+   * Creates a credential with quota project from environment if present
+   *
+   * @return credential with quota project from envoronment
+   */
+  public GoogleCredentials createWithQuotaProject() {
```

Comment:
This need to take the quota as a parameter and ADC should pass it from env. This should encapsulate checking the value and building the credential with it.

Reply:
Doing only null check. If user wants to set an empty value, they should be able to I think.